infrastructure/container_payload/run.py

```diff
@@ -174,113 +174,120 @@ def c(cl=[]):
-                if main:
-                    line = ['sudo', '-u', 'unprivileged', 'timeout', '10s',
-                            'bash', '-c',
-                            'LD_PRELOAD=/preloader.so {} {}'.format(
-                              os.path.join(workdir, main.split('.')[0]), cli)]
-                    c(line)
-
-        elif mode == "submit":
-            main = doctor_main_gpr(workdir, False)
+            if build() and main:
+                if mode == "run":
+                    # Check to see if cli.txt was sent from the front-end
+                    cli_txt = os.path.join(workdir, CLI_FILE)
+                    if os.path.isfile(cli_txt):
+                        cli = "`cat {}`".format(cli_txt);
+                        with open(cli_txt, 'r') as f:
+                            print("stdin: {}".format(f.read().replace('\n', ' ')))
+                    else:
+                        # otherwise pass no arguments to the main
+                        cli = ""
 
-            # In "submit" mode, first build, and then launch the main with test_cases
-            if c(["gprbuild", "-q", "-P", "main", "-gnatwa"]):
-                # Check to see if lab has IO resources
-                labio_txt = os.path.join(workdir, LAB_IO_FILE)
-                if os.path.isfile(labio_txt):
-                    # If it is found, read contents into string and replace
-                    #  newlines with spaces
-                    with open(labio_txt, 'r') as f:
-                        io_lines = f.readlines()
-
-                    # organize test instances
-                    test_cases = {}
-                    for line in io_lines:
-                        match = LAB_IO_REGEX.match(line)
-
-                        if match is not None:
-                            # found match(es)
-                            io = match.group(1)
-                            key = match.group(2)
-                            seq = match.group(3)
-
-                            if key in test_cases.keys():
-                                if io in test_cases[key].keys():
-                                    test_cases[key][io] += seq
+                    errno, stdout = run(main, workdir, cli)
+                    for line in stdout:
+                        print("stdout: {}".format(line))
+                else:
+                    # mode == "submit"
+                    # Check to see if lab has IO resources
+                    labio_txt = os.path.join(workdir, LAB_IO_FILE)
+                    if os.path.isfile(labio_txt):
+                        # If it is found, read contents
+                        with open(labio_txt, 'r') as f:
+                            io_lines = f.readlines()
+
+                        # organize test instances
+                        test_cases = {}
+                        for line in io_lines:
+                            match = LAB_IO_REGEX.match(line)
+
+                            if match is not None:
```

Comment:
(nit) I think you can just write "if match:" rather than "if match is not None:"